Tests/AEPOptimizeTests/UnitTests/OptimizePublicAPITests.swift

```diff
@@ -563,6 +560,455 @@ class OptimizePublicAPITests: XCTestCase {
+        // setup
+        let expectation = XCTestExpectation(description: "updatePropositions should dispatch an event with expected data.")
+        expectation.assertForOverFulfill = true
+
+        let testEventData: [String: Any] = [
+            "requesttype": "updatepropositions",
+            "surfaces": [
+                [ "name": "myView#featureJson"
+                ]
+            ],
+            "xdm": [
+                "myXdmKey": "myXdmValue"
+            ],
+            "data": [
+                "myKey": "myValue"
+            ]
+        ]
+        let testEvent = Event(name: "Optimize Update Propositions Request",
+                              type: "com.adobe.eventType.optimize",
+                              source: "com.adobe.eventSource.requestContent",
+                              data: testEventData)
+
+        // test
+        EventHub.shared.getExtensionContainer(MockExtension.self)?.registerListener(type: testEvent.type,
+                                                                                    source: testEvent.source) { event in
+            XCTAssertEqual(testEvent.name, event.name)
+            XCTAssertEqual("updatepropositions", event.data?["requesttype"] as? String)
+            guard let surfaces = event.data?["surfaces"] as? [String], !surfaces.isEmpty else {
+                XCTFail("Surface string array should be valid.")
+                return
+            }
+            XCTAssertEqual(1, surfaces.count)
+            XCTAssertEqual("myView#featureJson", surfaces[0])
+
+            guard let xdm = event.data?["xdm"] as? [String: Any] else {
+                XCTFail("Xdm data should be valid.")
+                return
+            }
+            XCTAssertEqual(1, xdm.count)
+            XCTAssertEqual("myXdmValue", xdm["myXdmKey"] as? String)
+
+            guard let data = event.data?["data"] as? [String: Any] else {
+                XCTFail("Freeform data should be valid.")
+                return
+            }
+            XCTAssertEqual(1, data.count)
+            XCTAssertEqual("myValue", data["myKey"] as? String)
+
+            expectation.fulfill()
+        }
+
+        // test
+        Optimize.updatePropositionsForSurfacePaths(["myView#featureJson"],
+                                    withXdm: ["myXdmKey": "myXdmValue"] as [String: Any],
+                                    andData: ["myKey": "myValue"] as [String: Any])
+
+        // verify
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func testUpdatePropositionsForSurfacePaths_multipleValidSurfaces() {
+        // setup
+        let expectation = XCTestExpectation(description: "updatePropositions should dispatch an event with expected data.")
+        expectation.assertForOverFulfill = true
+
+        let testEventData: [String: Any] = [
+            "requesttype": "updatepropositions",
+            "surfaces": [
+                [
+                    "name": "myView/mySubview1"
```

Comment:
Same question here. Is it supposed to be `"surfaces" : ["myView/mySubview1", ""myView/mySubview2"]`

Reply:
Same response as above.